guides/documentation/service-interceptors.md

```diff
@@ -149,6 +149,39 @@ functions of further interceptors in the path are invoked as if no
 pause had occurred. A single context may pause and resume an arbitrary
 number of times.
 
+### Examples
+
+Here's how you define a "before" interceptor:
+
+```clojure
+(ns interceptor.example
+  (:require [io.pedestal.interceptor :refer [interceptor]]))
+
+(def my-before-interceptor
+  (interceptor
+   {:name ::hello-world
+    :enter
+    (fn [context]
+      (assoc context :response 
+                     {:status 200 :body "Hello world!" 
+                      :headers {"Content-Type" "text/plain"}}))}))
```

Comment:
Hi @cap10morgan,

We've used the built-in helpers a lot in our codebase for defining interceptors. I wonder if it'd be helpful to others if we mention these functions and macros?

``` clj
(require '[io.pedestal.interceptor.helpers :as interceptor])

(def my-before-interceptor
  (interceptor/before
    ::my-before-interceptor
    (fn [context]
      (assoc-in context [:request :example] true))))
```

Although I've avoided use of the macro forms to date, there are helpers that call `def` for you too. This snippet is copied out of a docstring in the helpers namespace:

``` clj
(defbefore parse-body-as-wibblefish
  [context]
  (let [request (:request context)
        new-request (assoc request :wibblefish-params
                           (wibblefish-parse (:body request)))]
    (assoc context :request new-request)))
```

https://github.com/pedestal/pedestal/blob/c6e4a1f7e9c411cc83305ddcc451a954ab931ddc/service/src/io/pedestal/interceptor/helpers.clj